That's what I got in the <s>master</s>main and on this branch with ``./python -m test -T test_doctest``:
```
Using random seed: 3478682052
0:00:00 load avg: 0.96 Run 1 test sequentially
0:00:00 load avg: 0.96 [1/1] test_doctest
test_doctest passed in 38.9 sec

== Tests result: SUCCESS ==

1 test OK.

Total duration: 38.9 sec
Total tests: run=63
Total test files: run=1/1
Result: SUCCESS
Not printing coverage data for '/tmp/tmp7k4z8ing/doctest_testpkg/__init__.py': [Errno 2] No such file or directory: '/tmp/tmp7k4z8ing/doctest_testpkg/__init__.py'
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/home/sk/src/cpython/Lib/test/__main__.py", line 2, in <module>
    main(_add_python_opts=True)
  File "/home/sk/src/cpython/Lib/test/libregrtest/main.py", line 665, in main
    Regrtest(ns, _add_python_opts=_add_python_opts).main(tests=tests)
  File "/home/sk/src/cpython/Lib/test/libregrtest/main.py", line 657, in main
    exitcode = self.run_tests(selected, tests)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/sk/src/cpython/Lib/test/libregrtest/main.py", line 494, in run_tests
    return self._run_tests(selected, tests)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/sk/src/cpython/Lib/test/libregrtest/main.py", line 474, in _run_tests
    self.finalize_tests(tracer)
  File "/home/sk/src/cpython/Lib/test/libregrtest/main.py", line 374, in finalize_tests
    results.write_results(show_missing=True, summary=True,
  File "/home/sk/src/cpython/Lib/trace.py", line 273, in write_results
    with open(filename, 'rb') as fp:
         ^^^^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmp7k4z8ing/doctest_testpkg/__init__.py'
```
After removing problem doctest (on this branch)
<details>

```diff
diff --git a/Lib/test/test_doctest.py b/Lib/test/test_doctest.py
index e51f54ef35..72f49a6834 100644
--- a/Lib/test/test_doctest.py
+++ b/Lib/test/test_doctest.py
@@ -2843,39 +2843,6 @@ def test_lineendings(): r"""
     TestResults(failed=0, attempted=1)
     >>> os.remove(fn)
 
-Now we test with a package loader that has a get_data method, since that
-bypasses the standard universal newline handling so doctest has to do the
-newline conversion itself; let's make sure it does so correctly (issue 1812).
-We'll write a file inside the package that has all three kinds of line endings
-in it, and use a package hook to install a custom loader; on any platform,
-at least one of the line endings will raise a ValueError for inconsistent
-whitespace if doctest does not correctly do the newline conversion.
-
-    >>> from test.support import os_helper
-    >>> import shutil
-    >>> dn = tempfile.mkdtemp()
-    >>> pkg = os.path.join(dn, "doctest_testpkg")
-    >>> os.mkdir(pkg)
-    >>> os_helper.create_empty_file(os.path.join(pkg, "__init__.py"))
-    >>> fn = os.path.join(pkg, "doctest_testfile.txt")
-    >>> with open(fn, 'wb') as f:
-    ...     f.write(
-    ...         b'Test:\r\n\r\n'
-    ...         b'  >>> x = 1 + 1\r\n\r\n'
-    ...         b'Done.\r\n'
-    ...         b'Test:\n\n'
-    ...         b'  >>> x = 1 + 1\n\n'
-    ...         b'Done.\n'
-    ...         b'Test:\r\r'
-    ...         b'  >>> x = 1 + 1\r\r'
-    ...         b'Done.\r'
-    ...     )
-    95
-    >>> with test_hook(dn):
-    ...     doctest.testfile("doctest_testfile.txt", package="doctest_testpkg", verbose=False)
-    TestResults(failed=0, attempted=3)
-    >>> shutil.rmtree(dn)
-
 """
 
 def test_testmod(): r"""
```
</details>

I got
```
$ ./python -m test -T test_doctest | grep doctest
0:00:00 load avg: 0.44 [1/1] test_doctest
test_doctest passed in 38.3 sec
 1092    69%   doctest   (/home/sk/src/cpython/Lib/doctest.py)
    5    80%   test.doctest_aliases   (/home/sk/src/cpython/Lib/test/doctest_aliases.py)
   17    82%   test.doctest_lineno   (/home/sk/src/cpython/Lib/test/doctest_lineno.py)
   13   100%   test.sample_doctest   (/home/sk/src/cpython/Lib/test/sample_doctest.py)
    3    66%   test.sample_doctest_no_docstrings   (/home/sk/src/cpython/Lib/test/sample_doctest_no_docstrings.py)
    5    80%   test.sample_doctest_no_doctests   (/home/sk/src/cpython/Lib/test/sample_doctest_no_doctests.py)
  184    80%   test.test_doctest   (/home/sk/src/cpython/Lib/test/test_doctest.py)
Warning -- Unraisable exception
Exception ignored in: <module 'threading' from '/home/sk/src/cpython/Lib/threading.py'>
Traceback (most recent call last):
  File "/home/sk/src/cpython/Lib/threading.py", line 1588, in _shutdown
    assert tlock.locked()
AssertionError: 
```
So, at least we have some coverage for the doctest module.

With coveragepy:
```
python -m coverage run --branch --source=doctest Lib/test/regrtest.py test_doctest
0:00:00 load avg: 1.33 Run 1 test sequentially
0:00:00 load avg: 1.33 [1/1] test_doctest

== Tests result: SUCCESS ==

1 test OK.

Total duration: 7.0 sec
Total tests: run=59
Total test files: run=1/1
Result: SUCCESS
Warning -- Unraisable exception
Exception ignored in: <module 'threading' from '/home/sk/src/cpython/Lib/threading.py'>
Traceback (most recent call last):
  File "/home/sk/src/cpython/Lib/threading.py", line 1588, in _shutdown
    assert tlock.locked()
AssertionError: 
$ python -m coverage report
Name             Stmts   Miss Branch BrPart  Cover
--------------------------------------------------
Lib/doctest.py     983    136    448     64    84%
--------------------------------------------------
TOTAL              983    136    448     64    84%
Exception ignored in: <module 'threading' from '/home/sk/src/cpython/Lib/threading.py'>
Traceback (most recent call last):
  File "/home/sk/src/cpython/Lib/threading.py", line 1588, in _shutdown
    assert tlock.locked()
AssertionError: 
```

Apparently, we have some coverage statistics for this module.  (And few new bugs...)

--- a/Lib/test/test_doctest.py
+++ b/Lib/test/test_doctest.py
@@ -3219,7 +3219,4 @@ def load_tests(loader, tests, pattern):
 
 
 if __name__ == '__main__':
-    raise RuntimeError(
-        "Running `test_doctest` via cmdline is not supported, "
-        "use `-m test test_doctest` instead"
-    )
+    unittest.main(module='test.test_doctest')

